### PR TITLE
0.0.17

### PR DESCRIPTION
**Pull request by jamesplease**
Almost ready for release...just one thing I'm curious to get your thoughts on @LBRDan . Thank you _so_ much for adding this feature, it's an awesome addition!

### CODE REVIEW

File: src/lrud-input/bubble-key-input.ts [hunk 1]
```diff
@@ -58,10 +58,12 @@ export default function bubbleKey(focusTree: FocusStore, key: LRUDKey) {
       propagationStopped = true;
     }
 
-    const sourcingLeafId = focusH[0];
-    const targetNode: Node | undefined = state.nodes?.[sourcingLeafId];
+    const targetNodeId = reverseFocusHierarchy[0];
 
-    focusH.forEach((focusedNodeId) => {
+    // This is the equivalent of event.target within DOM events.
+    const targetNode: Node = state.nodes[targetNodeId] as Node;
```
Comment (jamesplease) [inline src/lrud-input/bubble-key-input.ts:64]: @LBRDan I noticed you had two defensive things going on here: checking that `state.nodes` is defined and that the returned node could be `undefined`. I think it's safe to assume that both exist, but I'm curious what you think.

File: README.md [hunk 2]
```diff
@@ -377,7 +377,8 @@ An object that is passed to you in the LRUD-related callbacks of a [`FocusNode`
 | ----------------- | ----------------------- | ---------------------------------------------------------------------------------------------------------------------- |
 | `key`             | string                  | A string representing the key that was pressed. One of `"left"`, `"right"`, `"up"`, `"down"`, `"select"`, or `"back"`. |
 | `isArrow`         | boolean                 | Whether or not this key is an arrow.                                                                                   |
-| `node`            | [FocusNode](#focusnode) | The [`FocusNode`](#focusnode) that received this event.                                                                |
+| `node`            | [FocusNode](#focusnode) | The current [`FocusNode`](#focusnode) that received this event as the event propagates up the focus hierarchy. Analagous to [`event.currentTarget`](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget) |
+| `targetNode`      | [FocusNode](#focusnode) | The leaf [`FocusNode`](#focusnode) from which this event propagated. Analagous to [`event.target`](https://developer.mozilla.org/en-US/docs/Web/API/Event/target) |
```
Comment (jamesplease) [inline README.md:381]: @LBRDan lmk if you have any thoughts on these docs ✌️

Comment (LBRDan) [reply README.md:381]: @jamesplease I'm so sorry, the docs addition slipped through my mind 
Looks perfect to me, thanks!

File: src/lrud-input/bubble-key-input.ts (hunk 1 above)
Comment (LBRDan) [reply src/lrud-input/bubble-key-input.ts:64]: @jamesplease You’re right
As most of the current solutions for the spatial navigation problem, this one conceptually assumes that there is always a leaf node
The guard is added more for a runtime getState() call is executed before calling all the events on the node.

I've been too cautious about this maybe...

File: README.md (hunk 2 above)
Comment (jamesplease) [reply README.md:381]: No worries at all @LBRDan ! Teamwork 👊

**Comment by jamesplease** [comment]
Released as 0.0.17. Thanks a ton, @LBRDan !

Also, if you're interested in continuing to contribute more to this lib and have questions or would find it useful to do a quick walkthrough of the code, lmk and we could hop on a call sometime. No worries if not though, just a thought!

**Comment by LBRDan** [comment]
Sorry to bother @jamesplease 
Any chance of viewing this published on npm?

**Comment by jamesplease** [comment]
Oops, I thought I did publish it! My mistake.

Alright, it should be published now. Let me know if you're seeing it!

**Pull request merged**